apis/v1alpha1/gateway_types.go

```diff
@@ -483,47 +485,125 @@ type GatewayStatus struct {
 	//
 	// These addresses should all be of type "IPAddress".
 	//
-	// +required
+	// +optional
 	Addresses []GatewayAddress `json:"addresses"`
 
 	// Conditions describe the current conditions of the Gateway.
-	// +optional
+	//
+	// Implementations should prefer to express Gateway conditions
+	// using the `GatewayConditionType` and `GatewayConditionReason`
+	// constants so that operators and tools can converge on a common
+	// vocabulary to describe Gateway state.
+	//
+	// Known condition types are:
+	//
+	// * "Scheduled"
+	// * "Ready"
+	//
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +kubebuilder:default={{type: "Scheduled", status: "False", reason:"NotReconciled", message:"Waiting for controller", lastTransitionTime: "1970-01-01T00:00:00Z"}}
```

Comment:
I think "NotReconciled" may be a better reason for the default GatewayClass condition as well.

Reply:
Also, any reason for defining the default twice? Once here and then on line 32 where the GatewayStatus struct is used?

Reply:
I had to do the same thing for GatewayClass. If someone created a GatewayClass with a `{}` status, I believe the default here would be required. Unfortunately the defaulting logic is pretty basic here so it involves a bit of repetition.